prover/src/constraints/evaluator/lagrange.rs

```diff
@@ -0,0 +1,289 @@
+use air::{
+    Air, LagrangeConstraintsCompositionCoefficients, LagrangeKernelConstraints,
+    LagrangeKernelEvaluationFrame,
+};
+use alloc::vec::Vec;
+use math::{batch_inversion, FieldElement};
+
+use crate::{StarkDomain, TraceLde};
+
+/// Contains a specific strategy for evaluating the Lagrange kernel boundary and transition
+/// constraints where the divisors' evaluation is batched.
+///
+/// Specifically, [`batch_inversion`] is used to reduce the number of divisions performed.
+pub struct LagrangeKernelConstraintsBatchEvaluator<E: FieldElement> {
+    lagrange_kernel_constraints: LagrangeKernelConstraints<E>,
+    rand_elements: Vec<E>,
+}
+
+impl<E: FieldElement> LagrangeKernelConstraintsBatchEvaluator<E> {
+    /// Constructs a new [`LagrangeConstraintsBatchEvaluator`].
+    pub fn new<A: Air>(
+        air: &A,
+        lagrange_kernel_rand_elements: Vec<E>,
+        lagrange_composition_coefficients: LagrangeConstraintsCompositionCoefficients<E>,
+    ) -> Self
+    where
+        E: FieldElement<BaseField = A::BaseField>,
+    {
+        Self {
+            lagrange_kernel_constraints: LagrangeKernelConstraints::new(
+                air.context(),
+                lagrange_composition_coefficients,
+                &lagrange_kernel_rand_elements,
+            ),
```

Comment:
Why not use `air.get_lagrange_kernel_constraints()` here instead of `LagrangeKernelConstraints::new()`?

Reply:
Changed to `air.get_lagrange_kernel_constraints()`; however since it returns an `Option<>`, we have to use an `expect()`, which arguably is not as nice?